### PR TITLE
Base api3

### PR DESCRIPTION
**Pull request by maruf0011**

### CODE REVIEW

File: src/xERC20/contracts/XUltraLRT.sol [hunk 1]
```diff
@@ -220,12 +221,11 @@ contract XUltraLRT is
      * @param receiver The address of the receiver
      */
     function deposit(uint256 _amount, address receiver) public whenNotPaused onlyTokenDepositAllowed {
-        if (block.timestamp - lastPriceUpdateTimeStamp > maxPriceLag) revert XErrors.NotUpdatedPrice();
         if (_amount == 0) revert XErrors.InvalidAmount();
         if (receiver == address(0)) revert XErrors.InvalidReceiver();
-        if (sharePrice == 0) revert XErrors.InvalidSharePrice();
         if (address(baseAsset) == address(0)) revert XErrors.InvalidBaseAsset();
 
+        uint256 tokenPrice = getSharePrice();
```
Comment (tmoon) [inline src/xERC20/contracts/XUltraLRT.sol]: why are we renaming it? can we keep it the same i.e. sharePrice?

File: src/xERC20/contracts/XUltraLRT.sol [hunk 2]
```diff
@@ -239,10 +239,30 @@ contract XUltraLRT is
         // accrued fees
         accruedFees += fees;
         // mint token
-        uint256 mintAmount = ((10 ** decimals()) * assetsToMintShares) / sharePrice;
+        uint256 mintAmount = ((10 ** decimals()) * assetsToMintShares) / tokenPrice;
```
Comment (tmoon) [inline src/xERC20/contracts/XUltraLRT.sol]: let's not rename, not worth it

**Review by tmoon — COMMENTED** [review]
let's try to keep the original contracts as unchanged as possible

File: src/xERC20/contracts/XUltraLRT.sol (hunk 2 above)
Comment (maruf0011) [reply src/xERC20/contracts/XUltraLRT.sol]: renamed to `_sharePrice`

**Pull request merged**